lib/rubocop/cop/rspec/verified_double_reference.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for consistent verified double reference style.
+      #
+      # @see https://relishapp.com/rspec/rspec-mocks/docs/verifying-doubles
+      #
+      # This cop can be configured in your configuration using the
+      # `EnforcedStyle` option and supports `--auto-gen-config`.
+      #
+      # @example `EnforcedStyle: constant`
+      #   # bad
+      #   let(:foo) do
+      #     instance_double('ClassName', method_name: 'returned_value')
+      #   end
+      #
+      #   # good
+      #   let(:foo) do
+      #     instance_double(ClassName, method_name: 'returned_value')
+      #   end
+      #
+      # @example `EnforcedStyle: string`
+      #   # bad
+      #   let(:foo) do
+      #     instance_double(ClassName, method_name: 'returned_value')
+      #   end
+      #
+      #   # good
+      #   let(:foo) do
+      #     instance_double('ClassName', method_name: 'returned_value')
+      #   end
+      class VerifiedDoubleReference < Base
+        extend AutoCorrector
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<style>s` references for verified doubles.'
+
+        RESTRICT_ON_SEND = %i[
+          class_double
+          class_spy
+          instance_double
+          instance_spy
+          mock_model
+          object_double
+          object_spy
+          stub_model
+        ].freeze
+
+        # @!method verified_double(node)
+        def_node_matcher :verified_double, <<-PATTERN
+          (send nil? _ $_ ...)
+        PATTERN
+
+        def on_send(node)
+          verified_double(node) do |class_reference|
+            return correct_style_detected if preferred_style?(class_reference)
+
+            message = format(MSG, style: style)
+            expression = class_reference.loc.expression
+
+            add_offense(expression, message: message) do |corrector|
+              violation = class_reference.children.last.to_s
+              corrector.replace(expression, correct_style(violation))
+            end
```

Comment:
Adding https://github.com/rubocop/rubocop/blob/1e55b1aa5e4c5eaeccad5d61f08b7930ed6bc341/lib/rubocop/cop/internal_affairs/style_detected_api_use.rb#L42 would pacify the internal cop.